Google.Cloud.EntityFrameworkCore.Spanner/Storage/Internal/SpannerRetriableTransaction.cs

```diff
@@ -286,17 +288,29 @@ internal async Task RetryAsync(SpannerException abortedException, CancellationTo
         /// <returns>Returns the UTC timestamp when the data was written to the database.</returns>
         public async Task<DateTime> CommitAsync(CancellationToken cancellationToken = default)
         {
+            var tracer = TracerProviderExtension.GetTracer();
+            using var span = tracer.StartActiveSpan(TracerProviderExtension.SPAN_NAME_COMMIT);
             while (true)
             {
                 try
                 {
                     _commitTimestamp = await SpannerTransaction.CommitAsync(cancellationToken).ConfigureAwait(false);
+                    span.SetStatus(OpenTelemetry.Trace.Status.Ok);
                     return (DateTime)_commitTimestamp;
                 }
                 catch (SpannerException e) when (e.ErrorCode == ErrorCode.Aborted)
                 {
                     await RetryAsync(e, cancellationToken).ConfigureAwait(false);
                 }
+                catch (Exception e)
+                {
+                    span.SetStatus(OpenTelemetry.Trace.Status.Error.WithDescription(e.Message));
+                    throw;
+                }
+                finally
+                {
+                    span.End();
```

Comment:
This will potentially be called multiple times if the transaction is retried for the same span, as the span is created outside the `while(true)` loop. I think it is a good choice to create the span outside the loop, as we would want to measure the total time that is needed to commit (including the retry), so:
1. The span should only be ended once the commit (including retries) has finished.
2. I think it would be good to also set the status `Retrying` if the commit must retry. That would be valuable information for any user investigating why some commits take longer than others.

Reply:
Done!